File: cbt-api/entity/jawaban_soal.go
package entity

import "time"

type JawabanSoal struct {
    IdJawabanSoal uint64    `gorm:"primary_key;autoIncrement" json:"id_jawaban_soal"`
    Jawaban       string    `gorm:"type:varchar(255);not null" json:"jawaban"`
    Benar         bool      `gorm:"type:tinyint(1);not null" json:"benar"`
    IdSoal        uint64    `gorm:"type:bigint(20) unsigned" json:"id_soal"`
    IdTipeSoal    uint64    `gorm:"type:bigint(20) unsigned" json:"id_tipe_soal"`
    CreatedAt     time.Time `gorm:"type:timestamp;default:CURRENT_TIMESTAMP" json:"created_at"`
    UpdatedAt     time.Time `gorm:"type:timestamp;default:CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP" json:"updated_at"`

    Soal      Soal      `gorm:"foreignkey:IdSoal;constraint:onUpdate:CASCADE, onDelete:CASCADE" json:"soal"`
    TipeSoal  TipeSoal  `gorm:"foreignkey:IdTipeSoal;constraint:onUpdate:CASCADE, onDelete:CASCADE" json:"tipe_soal"`
}

File: computer-based-test/app/Models/jawaban_soal.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class jawaban_soal extends Model
{

    protected $table = 'jawaban_soal'; 

    protected $primaryKey = 'id_jawaban_soal';

    protected $fillable = [
        'id_jawaban_siswa',
        'jawaban',
        'benar',
        'id_soal',
        'id_tipe_soal',
    ];

    public function soal(){
        return $this->belongsTo(soal::class);
    }

    public function tipe_soal(){
        return $this->belongsTo(tipe_soal::class);
    }

    public function jawaban_siswa(){
        return $this->hasMany(jawaban_siswa::class);
    }
}
